### ISSUE TITLE
Finding a new home for RDPWrap

### ISSUE THREAD

**Issue by fragtion** [fre4kyC0de/rdpwrap#40]
In the interest of protecting fre4kyC0de's account (as he/she represents all of us here and it would be very unfortunate for him/her to take the brunt of punishment for something that we all have a common interest in), I feel like this should be discussed immediately while we still have an opportunity to do so

As most of you will already know, Microsoft have owned github for some time already, and more recently set a precedent by nuking stascorp's rdpwrap repo (https://github.com/stascorp/rdpwrap). They have therefore made their intentions clear, and personally I am frankly surprised that this didn't happen earlier, given their staunch track record with regards to their stances pn licensing, piracy, and the like

Based on that, it should therefore seem apparent enough to all of us, that it will only be an imminent matter of time before they start banning any alternative forks and references to the rdpwrap project more aggressively in the near future, and so I fear that if we do not take appropriate actions now, this could disperse our community and leave us stranded with no central point of confluence for discussion and further development...

To Microsoft's credit, I think we can all agree that the actions they have taken are rationally within right and reason, and should have been anticipated by us, as it is principally obscure for them to permit the hosting of a project on one of their own platforms which in some way undermines the integrity of their products from a licensing perspective

That being said - all of us, Microsoft included, should know and be able to concede to the fact that "the cat is already out of the bag" when it comes to the art of terminal services patching, just as it is (and has always been) the case with Activation of Windows in general. While RDPWrap may be the most prominent project in its class, it is by no means the only one, and unless/until Microsoft were to radically redesign and transform their terminal services libraries from a security perspective, then it cannot reasonably be doubted or denied that further development from other smart minds who prize liberty above all else, is simply inevitable

The writing is therefore on the wall, that if this project is to ever survive the mighty wrath of Microsoft's insatiable desire for control, it will at the very least need to find a new home elsewhere. I personally believe that we will be able to continue as we have been, and that Microsoft may largely be willing to turn a blind eye (albeit with some raised brows), so long as we take the project off of github - as them hosting it here until now clearly presented an awkward dilemma that they were simply not prepared to sustain.

It is unfortunate that it has to be this way, as Github should really be a neutral haven for software development that should be free from the tentacles of biased corporate policies of what should essentially be seen as another company with distinct terms and policies altogether, but alas the realization of such ideals is clearly a matter that is beyond our control and influence, apart from the fact that the community will need to "vote" with our choice as to whom we support with hosting our projects going forward

As for where we go to from here, that remains somewhat of a mystery I suppose. Gitlab seems to one obvious contender. Another option is to self-host a git-scm repo on private webspace... But anyway that's where all of us that together comprise this community will all need to play our part where we can, if we really care about the future of this project and ensuring that it lives on beyond the barring from github

Hoping to hear some of your suggestions. The sooner we can relocate a new and secure home for this project elsewhere, the better for everyone involved or interested, I'm sure

**Comment by greasemonkey2** [comment]
+1

**Comment by malicon** [comment]
**fragtion** Very good presentation. Do you have any suggestions for a new source hosting facility? Many years ago I used Source Forge which was very popular at the time until it was purchased by BizX LLC which introduced bundling of ad-ware and potentially unwanted programs (PUP) to the project downloads. This approach has been strongly criticized and has surely helped Github gain popularity.  Apparently they no longer do that(?), though SF post some adds in the project pages.  So SF could be a candidate.  Wikipedia in https://en.wikipedia.org/wiki/Comparison_of_source-code-hosting_facilities list a bunch of source hosting providers.  We just have to pick a FREE good one that won't complain about the nature of the RDP Wrapper project...

But, since we need someone willing to be responsible of the project, that person (or persons) should choose what tool/provider he/they want.  stas'M corp (see stascorp.com) who was the author of the first versions of RDP Wrapper library and maybe was till the Github staff locked the repository might be interested.  So I hope he/she will raise his hand somewhere here. 

We also need the latest possible sources.  May be stascorp or who ever is behind the latest source modifications will be kind enough to provide those sources even he/they don't want to be involved in the maintenance of the new repository. If the author(s) don't show up, obviously there are people out there that know much more then me about RDP Wrapper library.  I mean those great people that hand us appropriate entries to add to the rdpwrap.ini file to support the new termsrv.dll versions MS pushes.  Chances are that they have a recent source code copy on their PC.  Please tell us what you have.

**Comment by devfin** [comment]
а мо

> TL;DR? Read the title & offer any suggestions if you have any.
> 
> In the interest of protecting fre4kyC0de's account (as he/she represents all of us here and it would be very unfortunate for him/her to take the brunt of punishment for something that we all have a common vested interest in), I feel like this should be discussed immediately while we still have an opportunity to do so..
> 
> As most of you will already know, Microsoft have owned github for some time already, and more recently set a precedent by nuking stascorp's rdpwrap repo (https://github.com/stascorp/rdpwrap). They have therefore made their intentions clear, and personally I am frankly surprised that this didn't happen earlier, given their staunch track record with their stances on licensing, piracy, and the like
> 
> Based on that, it should therefore seem apparent enough to all of us, that it will only be an imminent matter of time before they start banning any alternative forks and references to the rdpwrap project more aggressively in the near future, and so I fear that if we do not take appropriate actions now, this could disperse our community and leave us stranded with no central point of confluence for discussion and further development...
> 
> To Microsoft's credit, I think we can all agree that the actions they have taken are rationally within right and reason, and should have been anticipated by us, as it is principally obscure for them to permit the hosting of a project on one of their own platforms which in some way goes to violate/undermine the integrity of another of their products from a licensing terms perspective
> 
> That being said - all of us, Microsoft included, should know and be able to concede to the fact that "the cat is already out of the bag" when it comes to the art of terminal services patching, just as it is (and has always been) the case with Activation of Windows in general. While RDPWrap may be the most prominent project in its class, it is by no means the only one, and unless/until Microsoft were to radically redesign and transform their terminal services libraries from a security standpoint, then it cannot plausibly be doubted or denied that further development in this field from other smart minds who too prize personal liberty above all else and are passionately hellbent in that regard, is simply inevitable
> 
> The writing is therefore on the wall, that if this project is to ever survive the mighty wrath of Microsoft's insatiable desire for control, it will at the very least need to find a new home elsewhere. I personally believe that Microsoft will largely turn a blind eye to the continued development of this project (albeit with some raised brows), so long as it is removed from Github, as them hosting it here until now clearly presented an awkward dilemma that they were simply not prepared to sustain, as their reluctance to ban the project might have been construed by some as a kind of implicit endorsement, perhaps?
> 
> It is unfortunate that it has to be this way, as Github should really be a neutral haven for software development that should be free from the tentacles of biased corporate policies of what should essentially be seen as another company with distinct terms and policies altogether, but alas the realization of such ideals is clearly a matter that is beyond our control and influence, apart from the fact that the community will need to "vote" with our choice as to whom we support with hosting our own projects going forward
> 
> As for where we go to from here, that remains somewhat of a mystery I suppose. Gitlab seems to be one obvious contender. Another option may be to self-host a git-scm repo on private webspace, although that too could open a can of worms with regards to risk and liability (akin to thepiratebay, et al)... But anyway that's where all who comprise this community will need to play their part if we really care about the future of this project and ensuring that it lives on beyond the barring from github
> 
> I think we can all agree that the sooner this project is relocated to a secure new home elsewhere, the better, and that this is realistically our only option forward at this point. Hoping to hear some suggestions...! Apologies for a bit of a wall of text, but I wanted to cover this subject in as holistically comprehensive way as possible.

maybe Microsoft sees a potential hole to crack the system?

**Comment by fragtion** [comment]
> maybe Microsoft sees a potential hole to crack the system?
What do you mean exactly ?

**Comment by devfin** [comment]
I don’t have any information to whom the rdpwapper can still give access to my windows system, how much is the rdpwapper protected from hacking by intruders

**Comment by fre4kyC0de** [comment]
I can understand, that isn't liked by MS; that's why I don't know, if I should continue with this repo...

To prevent the distribution of Windows files, I'll make this repo read-only until I decided, what to do.
Maybe this repo will be open again, maybe I'll delete it. Until that point, I pushed the updates for 2020-05.
